apps/rig_inbound_gateway/lib/rig_inbound_gateway_web/v1/longpolling_controller.ex

```diff
@@ -0,0 +1,80 @@
+defmodule RigInboundGatewayWeb.V1.LongpollingController do
+  @moduledoc """
+  Handling of Longpolling functionalitiy
+  """
+  require Logger
+  use Rig.Config, [:cors]
+
+  use RigInboundGatewayWeb, :controller
+
+  alias RigOutboundGateway
+  alias RigInboundGatewayWeb.Session
+  alias Rig.Connection
+
+  @doc false
+  def handle_connection(%{method: "GET"} = conn, _params) do
+    conn = conn |> fetch_cookies
+
+    is_new_session =
+      if conn.req_cookies["connection_token"] === nil do
+        true
+      else
+        {:ok, session_pid} = Connection.Codec.deserialize(conn.req_cookies["connection_token"])
+        !Process.alive?(session_pid)
+      end
+
+    # No session set -> start new session
+    if(is_new_session) do
```

Comment:
if/else expressions this long should go into a function that's named after what this is deciding and dispatch using pattern matching, like this:

```elixir
defp what_this_decides(is_new_session)
defp what_this_decides(true) do
  ...
end
defp what_this_decides(false) do
  ...
end
```